components/resourceDetails/resourceSharing/agentAccessList/index.jsx

```diff
@@ -19,109 +19,67 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-/* eslint-disable camelcase */
-
-import React from "react";
-import {
-  Avatar,
-  createStyles,
-  List,
-  ListItem,
-  Typography,
-} from "@material-ui/core";
-import { makeStyles } from "@material-ui/styles";
+import React, { useContext, useEffect, useState } from "react";
 import T from "prop-types";
-import { useSession } from "@inrupt/solid-ui-react";
-import { displayProfileName } from "../../../../src/solidClientHelpers/profile";
-import styles from "../../styles";
-import PermissionsForm from "../../../permissionsForm";
+import { CircularProgress, List, ListItem } from "@material-ui/core";
+import { createStyles, makeStyles } from "@material-ui/styles";
+import { DatasetContext, useSession } from "@inrupt/solid-ui-react";
+import { Button } from "@inrupt/prism-react-components";
+import styles from "./styles";
+import AgentAccess from "../agentAccess";
+import { getPermissions } from "../../../../src/solidClientHelpers/permissions";
 
 const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 
-const TESTCAFE_ID_AGENT_WEB_ID = "agent-web-id";
-
-export function handleSave({
-  onSubmit,
-  saveFn,
-  onSave,
-  profile,
-  iri,
-  webId,
-  fetch,
-}) {
-  return async (access) => {
-    onSubmit(profile, access);
+function AgentAccessList({ onLoading }) {
+  const classes = useStyles();
+  const { fetch } = useSession();
+  const [permissions, setPermissions] = useState([]);
+  const [showAll, setShowAll] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const { dataset } = useContext(DatasetContext);
 
-    const response = await saveFn({
-      iri,
-      webId,
-      access,
-      fetch,
+  useEffect(() => {
+    if (!dataset) return;
+    setLoading(true);
+    getPermissions(dataset, fetch).then((normalizedPermissions) => {
```

Comment:
Rather than using `getPermissions` directly, you should create a hook similar to the ones in https://github.com/inrupt/pod-browser/blob/master/src/hooks/solidClient/index.js and use `swr` to improve loading times.

Reply:
I'm having trouble extracting this logic into a hook =/

Specifically because of the if-forks in https://github.com/inrupt/pod-browser/blob/design/sharing-details/src/solidClientHelpers/permissions.js#L190-L198, and into the function that fetches the profile (https://github.com/inrupt/pod-browser/blob/design/sharing-details/src/solidClientHelpers/profile.js#L52-L61).